crates/iota-indexer/src/main.rs

```diff
@@ -19,26 +58,180 @@ async fn main() -> Result<(), IndexerError> {
+    let opts = match iota_indexer::config::IndexerConfig::try_parse() {
+        Ok(opts) => opts,
+        Err(e) => {
+            e.print()
+                .map_err(|e| IndexerError::Generic(format!("Failed writing clap error: {e}")))?;
+            warn!(
+                "Parsing arguments using new CLI failed. Falling back to the old CLI, note that this will be deprecated in the future."
+            );
+            let mut old_conf = OldIndexerConfig::from_arg_matches_mut(
+                &mut OldIndexerConfig::command().version(VERSION).get_matches(),
+            )
+            .map_err(|e| {
+                IndexerError::Generic(format!("Failed parsing arguments using old CLI: {e}"))
+            })?;
+
+            // TODO: Explore other options as in upstream.
+            // For the moment we only use the fullnode for fetching checkpoints
+            old_conf.remote_store_url = Some(format!("{}/api/v1", old_conf.rpc_client_url));
+
+            let db_url = old_conf.get_db_url();
+
+            // NOTE: this parses the input host addr and port number for socket addr,
+            // so unwrap() is safe here.
+            let metrics_address = format!(
+                "{}:{}",
+                old_conf.client_metric_host, old_conf.client_metric_port
+            )
+            .parse()
+            .unwrap();
+
+            let download_queue_size = std::env::var("DOWNLOAD_QUEUE_SIZE")
+                .unwrap_or_else(|_| {
+                    IngestionConfig::DEFAULT_CHECKPOINT_DOWNLOAD_QUEUE_SIZE.to_string()
+                })
+                .parse::<usize>()
+                .expect("Invalid DOWNLOAD_QUEUE_SIZE");
+            let ingestion_reader_timeout_secs = std::env::var("INGESTION_READER_TIMEOUT_SECS")
+                .unwrap_or_else(|_| {
+                    IngestionConfig::DEFAULT_CHECKPOINT_DOWNLOAD_TIMEOUT.to_string()
+                })
+                .parse::<u64>()
+                .expect("Invalid INGESTION_READER_TIMEOUT_SECS");
+            let data_limit = std::env::var("CHECKPOINT_PROCESSING_BATCH_DATA_LIMIT")
+                .unwrap_or(
+                    IngestionConfig::DEFAULT_CHECKPOINT_DOWNLOAD_QUEUE_SIZE_BYTES.to_string(),
+                )
+                .parse::<usize>()
+                .unwrap();
+
+            let snapshot_min_lag = std::env::var("OBJECTS_SNAPSHOT_MIN_CHECKPOINT_LAG")
+                .ok()
+                .and_then(|s| s.parse::<usize>().ok())
+                .unwrap_or(SnapshotLagConfig::DEFAULT_MIN_LAG);
+
+            let rpc_client_url_parsed = old_conf
+                .rpc_client_url
+                .parse()
+                .expect("RPC Client url should be valid");
+
+            let command = if old_conf.analytical_worker {
+                Command::AnalyticalWorker
+            } else if old_conf.reset_db {
+                Command::ResetDatabase { force: true }
```

Comment:
This is not mapping to the existing behaviour where resetting the db is not a standalone command.

I believe that we should just nest `reset-db` as an option to the `indexer` command in the new CLI as well.